citrination_client/data/client.py

```diff
@@ -0,0 +1,301 @@
+        in a given dataset.
+
+        :param dataset_id: The id of the dataset to retrieve files from
+        :type dataset_id: int
+        :param glob: A regex used to select one or more files in the dataset
+        :type glob: str
+        :param is_dir: Whether or not the supplied pattern should be treated as a directory to search in
+        :type is_dir: bool
+        :param version_number: The version number of the dataset to retrieve files from
+        :type version_number: int
+        :return: A list of dataset files whose paths match the provided pattern.
+        :rtype: list of :class:`DatasetFile`
+        """
+        if version_number is None:
+            latest = True
+        else:
+            latest = False
+
+        data = {
+            "download_request": {
+                "glob": glob,
+                "isDir": is_dir,
+                "latest": latest
+            }
+        }
+
+        failure_message = "Failed to get matched files in dataset {}".format(dataset_id)
+
+        versions = self._get_success_json(self._post_json(routes.matched_files(dataset_id), data, failure_message=failure_message))['versions']
+
+        # if you don't provide a version number, only the latest
+        # will be included in the response body
+        if version_number is None:
+            version = versions[0]
+        else:
+            try:
+                version = list(filter(lambda v: v['number'] == version_number, versions))[0]
+            except IndexError:
+                raise ResourceNotFoundException()
+
+        return list(
+            map(
+                lambda f: DatasetFile(path=f['filename'], url=f['url']), version['files']
+                )
+            )
+
+    def get_dataset_file(self, dataset_id, file_path, version = None):
+        """
+        Retrieves a dataset file matching a provided file path
+
+        :param dataset_id: The id of the dataset to retrieve file from
+        :type dataset_id: int
+        :param file_path: The file path within the dataset
+        :type file_path: str
+        :param version: The dataset version to look for the file in. If nothing is supplied, the latest dataset version will be searched
+        :type version: int
+        :return: A dataset file matching the filepath provided
+        :rtype: :class:`DatasetFile`
+        """
+        return self.get_dataset_files(dataset_id, "^{}$".format(file_path), version_number=version)[0]
+
+    def get_pif(self, dataset_id, uid, dataset_version = None):
+        """
+        Retrieves a PIF from a given dataset.
+
+        :param dataset_id: The id of the dataset to retrieve PIF from
+        :type dataset_id: int
+        :param uid: The uid of the PIF to retrieve
+        :type uid: str
+        :param dataset_version: The dataset version to look for the PIF in. If nothing is supplied, the latest dataset version will be searched
+        :type dataset_version: int
+        :return: A :class:`Pif` object
```

Comment:
This is because I haven't accepted the default sphinx autodoc rst formatting I think. The default formatting is very un-friendly and I'm not sure how  to make the links appear in it's absence. Perhaps we can tackle this in the future?